ubyssey/widgets.py

```diff
@@ -1,15 +1,45 @@
+from datetime import date
+
 from dispatch.theme import register
 from dispatch.theme.widgets import Widget
 from dispatch.theme.fields import CharField, EventField
-from zones import EventSidebar
+from dispatch.apps.events.models import Event
+
+from zones import HomePageSidebar, HomePageSidebarBottom
 
 @register.widget
-class EventWidget(Widget):
-  id = 'custom'
-  name = 'Event Widget'
-  template = 'widgets/event.html'
-  zones = (EventSidebar,)
+class UpcomingEvents(Widget):
+  id = 'upcoming-events'
+  name = 'Upcoming Events'
+  template = 'widgets/upcoming-events.html'
+  zones = (HomePageSidebar, HomePageSidebarBottom)
+
+  featured_event = EventField('Featured Events', many=False)
+  number_of_events = CharField('Number of Events')
+
+  def prepare_data(self):
```

Comment:
Now that the fix in https://github.com/ubyssey/dispatch/pull/621 is merged, you should be able to override `context` instead of `prepare_data`.